### PR TITLE
chore: update upgrade notes for 0.7

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

Docs for upgrading

### CODE REVIEW

File: docs-md/operate-and-deploy/installation/upgrading.md [hunk 1]
```diff
@@ -6,17 +6,307 @@ description: Learn how to upgrade your on-premises ksqlDB deployments.
...
 
-Upgrading to ksqlDB 0.6.0
--------------------------
+## Why does ksqlDB not currently support in-place upgrades?
 
-Notable changes in 0.6.0:
+Past releases of KSQL were backwards compatible. However, there was a cost to this backwards compatibility:
+progress was slower and the code base incurred increased complexity.  ksqlDB is a young product and
+we're wanting to move fast, so have decided to choose speed of development over strong backwards
+compatibility guarantees for a few released.
```
Comment (vpapavas) [inline docs-md/operate-and-deploy/installation/upgrading.md]: typo: released -> releases

File: docs-md/operate-and-deploy/installation/upgrading.md [hunk 2]
```diff
@@ -6,17 +6,307 @@ description: Learn how to upgrade your on-premises ksqlDB deployments.
...
+Upgrading a cluster involves leaving the old cluster running on the old version, bringing up a new
+cluster on the new version, porting across your database schema and finally thinking about your data.
+
+### Porting the database schema
+
+To port your database schema from one cluster to another you need to recreate all the streams,
+tables and types in the source cluster.  This is currently a manual process, until support is added
+to [dump the schema](https://github.com/confluentinc/ksql/issues/4529).
+
+The recommended process it to:
```
Comment (vpapavas) [inline docs-md/operate-and-deploy/installation/upgrading.md]: typo: it -> is

**Comment by apurvam** [comment]
I'm going to merge, since the build is not operable at the moment due to the nexus -> Jfrog migration. this is a docs only change, so we don't need to wait on a green build.

File: docs-md/operate-and-deploy/installation/upgrading.md [hunk 3]
```diff
@@ -6,17 +6,307 @@ description: Learn how to upgrade your on-premises ksqlDB deployments.
...
+1. Capture tables SQL:
+  1. Run `list tables extended;` to list all of the tables.
+  1. Grab the SQL statement that created each table from the output.
+1. Capture custom types SQL:
+  1. Run `list types;` to list all of the custom types.
+  1. Convert the output into `CREATE TYPE <name> AS <schema>` syntax by grabbing the name from the
+     first column and the schema from the second column of the output.
+1. Order by dependency:  you'll now have the list of SQL statements to rebuild the schema, but they
+   are not yet ordered in terms of dependencies. You will need to reorder the statements to ensure
+   each statement come after any other statements it depends on.
```
Comment (vpapavas) [inline docs-md/operate-and-deploy/installation/upgrading.md]: typo: come -> comes

**Pull request merged**